src/cc/frontends/clang/b_frontend_action.cc

```diff
@@ -216,10 +212,11 @@ ProbeVisitor::ProbeVisitor(ASTContext &C, Rewriter &rewriter,
 
 bool ProbeVisitor::VisitVarDecl(VarDecl *D) {
   if (Expr *E = D->getInit()) {
-    ProbeChecker checker = ProbeChecker(E, ptregs_, track_helpers_, true);
-    if (checker.is_transitive() || IsContextMemberExpr(E)) {
-      tuple<Decl *, int> pt = make_tuple(D, checker.get_nb_derefs());
-      set_ptreg(pt);
+    ProbeChecker checker = ProbeChecker(E, extregs_, track_helpers_, true);
+    if ((checker.needs_probe() && C.getTypeSize(E->getType()) == 64) ||
```

Comment:
What I mentioned above. Instead of calling `is_transitive()` we check that the expression has size 64 bits. It's easier to do here instead of inside of `ProbeChecker` since it requires a reference to ASTContext and is only necessary in a couple places.